src/Metadata/Resource/Factory/StaticResourceNameCollectionFactory.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Metadata\Resource\Factory;
+
+use ApiPlatform\Metadata\Resource\ResourceNameCollection;
+
+final class StaticResourceNameCollectionFactory implements ResourceNameCollectionFactoryInterface
```

Comment:
I don't like the "static" wording. Internal? Included?

Reply:
I think that this will get removed as its not used? It is kind of static as it declares classes within the ResourceNameCollectionFactory.